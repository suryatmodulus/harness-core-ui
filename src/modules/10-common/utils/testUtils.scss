/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
