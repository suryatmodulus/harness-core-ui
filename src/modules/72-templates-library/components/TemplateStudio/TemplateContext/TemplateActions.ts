/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { EntityGitDetails, EntityValidityDetails, NGTemplateInfoConfig } from 'services/template-ng'
import type { TemplateViewData } from '@templates-library/components/TemplateStudio/TemplateContext/TemplateReducer'
import type { YamlBuilderHandlerBinding } from '@common/interfaces/YAMLBuilderProps'

export enum TemplateActions {
  DBInitialize = 'DBInitialize',
  Initialize = 'Initialize',
  Fetching = 'Fetching',
  Loading = 'Loading',
  UpdateTemplateView = 'UpdateTemplateView',
  UpdateTemplate = 'UpdateTemplate',
  SetYamlHandler = 'SetYamlHandler',
  Success = 'Success',
  Error = 'Error'
}

export interface ActionResponse {
  error?: string
  isUpdated?: boolean
  template?: NGTemplateInfoConfig
  yamlHandler?: YamlBuilderHandlerBinding
  originalTemplate?: NGTemplateInfoConfig
  isBETemplateUpdated?: boolean
  templateView?: TemplateViewData
  stableVersion?: string
  versions?: string[]
  isLoading?: boolean
  gitDetails?: EntityGitDetails
  entityValidityDetails?: EntityValidityDetails
  templateYaml?: string
}

export interface ActionReturnType {
  type: TemplateActions
  response?: ActionResponse
}

const dbInitialized = (): ActionReturnType => ({ type: TemplateActions.DBInitialize })
const initialized = (): ActionReturnType => ({ type: TemplateActions.Initialize })
const loading = (response: ActionResponse): ActionReturnType => ({ type: TemplateActions.Loading, response })
const updateTemplateView = (response: ActionResponse): ActionReturnType => ({
  type: TemplateActions.UpdateTemplateView,
  response
})
const setYamlHandler = (response: ActionResponse): ActionReturnType => ({
  type: TemplateActions.SetYamlHandler,
  response
})
const updating = (): ActionReturnType => ({ type: TemplateActions.UpdateTemplate })
const fetching = (): ActionReturnType => ({ type: TemplateActions.Fetching })
const success = (response: ActionResponse): ActionReturnType => ({ type: TemplateActions.Success, response })
const error = (response: ActionResponse): ActionReturnType => ({ type: TemplateActions.Error, response })

export const TemplateContextActions = {
  dbInitialized,
  initialized,
  loading,
  updating,
  fetching,
  updateTemplateView,
  setYamlHandler,
  success,
  error
}
