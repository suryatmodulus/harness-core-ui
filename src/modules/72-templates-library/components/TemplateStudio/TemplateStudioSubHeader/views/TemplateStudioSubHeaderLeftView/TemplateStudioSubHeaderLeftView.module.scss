/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.createTemplateDialog {
  width: 875px;
  padding-bottom: 0;
  position: relative;
}

.subHeaderLeftView {
  .templateName {
    font-size: 20px !important;
    text-overflow: ellipsis;
    white-space: nowrap !important;
    max-width: 240px;
    overflow: hidden;
  }

  .versionDropDown {
    :global(.DropDown--dropdownButton) {
      background-color: transparent !important;
      border-color: var(--primary-6) !important;
      height: 30px;
      min-width: var(--spacing-11) !important;
      max-width: 120px;
      column-gap: var(--spacing-4) !important;
    }
    :global(.DropDown--label) {
      color: var(--primary-7) !important;
      font-size: 13px !important;
    }
    :global(.DropDown--btnWrapper) {
      svg {
        color: var(--primary-7);
      }
    }
  }
}
