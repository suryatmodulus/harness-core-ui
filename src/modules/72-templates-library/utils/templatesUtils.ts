import type { UseStringsReturn } from 'framework/strings'

export enum TemplateType {
  Step = 'Step',
  Stage = 'Stage',
  Pipeline = 'Pipeline',
  Service = 'Service',
  Infrastructure = 'Infrastructure',
  StepGroup = 'StepGroup',
  Execution = 'Execution'
}

export const AllTemplatesTypes = 'All'

export interface TemplateTypeOption {
  label: string
  value: string
  disabled?: boolean
}

export const getAllowedTemplateTypes = (getString: UseStringsReturn['getString']): TemplateTypeOption[] => [
  {
    label: getString('step'),
    value: TemplateType.Step
  },
  {
    label: getString('templatesLibrary.stageTemplate'),
    value: TemplateType.Stage
  },
  {
    label: getString('common.pipeline'),
    value: TemplateType.Pipeline,
    disabled: true
  },
  {
    label: getString('service'),
    value: TemplateType.Service,
    disabled: true
  },
  {
    label: getString('infrastructureText'),
    value: TemplateType.Infrastructure,
    disabled: true
  },
  {
    label: getString('stepGroup'),
    value: TemplateType.StepGroup,
    disabled: true
  },
  {
    label: getString('executionText'),
    value: TemplateType.Execution,
    disabled: true
  }
]
