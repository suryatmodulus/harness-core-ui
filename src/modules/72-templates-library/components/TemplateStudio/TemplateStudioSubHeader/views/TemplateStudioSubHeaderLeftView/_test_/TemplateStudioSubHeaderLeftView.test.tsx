/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { act, fireEvent, getByRole, render } from '@testing-library/react'
import { findDialogContainer, TestWrapper } from '@common/utils/testUtils'
import templateContextMock from '@templates-library/components/TemplateStudio/SaveTemplatePopover/_test_/stateMock'
import { TemplateContext } from '@templates-library/components/TemplateStudio/TemplateContext/TemplateContext'
import routes from '@common/RouteDefinitions'
import { accountPathProps, pipelineModuleParams, templatePathProps } from '@common/utils/routeUtils'
import { useUpdateStableTemplate } from 'services/template-ng'
import { TemplateStudioSubHeaderLeftView } from '../TemplateStudioSubHeaderLeftView'

jest.mock('@wings-software/monaco-yaml/lib/esm/languageservice/yamlLanguageService', () => ({
  getLanguageService: jest.fn()
}))

jest.mock('services/template-ng', () => ({
  useUpdateStableTemplate: jest.fn().mockImplementation(() => {
    return {
      mutate: () =>
        Promise.resolve({
          data: { name: 'NewConnectorCreated' }
        }),
      loading: false
    }
  })
}))

describe('<TemplateStudioSubHeaderLeftView /> tests', () => {
  test('snapshot test', async () => {
    const { container } = render(
      <TemplateContext.Provider value={templateContextMock}>
        <TestWrapper
          path={routes.toTemplateStudio({ ...accountPathProps, ...templatePathProps, ...pipelineModuleParams })}
          pathParams={{
            templateIdentifier: 'Test_Http_Template',
            accountId: 'accountId',
            orgIdentifier: 'default',
            projectIdentifier: 'Yogesh_Test',
            module: 'cd',
            templateType: 'Step',
            versionLabel: 'v1'
          }}
        >
          <TemplateStudioSubHeaderLeftView />
        </TestWrapper>
      </TemplateContext.Provider>
    )
    expect(container).toMatchSnapshot()
  })

  test('update stable template test', async () => {
    const { getByText } = render(
      <TemplateContext.Provider value={templateContextMock}>
        <TestWrapper
          path={routes.toTemplateStudio({ ...accountPathProps, ...templatePathProps, ...pipelineModuleParams })}
          pathParams={{
            templateIdentifier: 'Test_Http_Template',
            accountId: 'accountId',
            orgIdentifier: 'default',
            projectIdentifier: 'Yogesh_Test',
            module: 'cd',
            templateType: 'Step',
            versionLabel: 'v1'
          }}
        >
          <TemplateStudioSubHeaderLeftView />
        </TestWrapper>
      </TemplateContext.Provider>
    )

    const stableBtn = getByText('common.setAsStable')
    await act(async () => {
      fireEvent.click(stableBtn)
    })

    const modal = findDialogContainer()
    expect(modal).toMatchSnapshot()
    const confirmBtn = getByRole(modal!, 'button', { name: 'confirm' })
    expect(confirmBtn).toBeDefined()
    await act(async () => {
      fireEvent.click(confirmBtn)
    })
    expect(useUpdateStableTemplate).toBeCalled()
  })

  test('edit btn should work as expected', async () => {
    const { container } = render(
      <TemplateContext.Provider value={templateContextMock}>
        <TestWrapper
          path={routes.toTemplateStudio({ ...accountPathProps, ...templatePathProps, ...pipelineModuleParams })}
          pathParams={{
            templateIdentifier: 'Test_Http_Template',
            accountId: 'accountId',
            orgIdentifier: 'default',
            projectIdentifier: 'Yogesh_Test',
            module: 'cd',
            templateType: 'Step',
            versionLabel: 'v1'
          }}
        >
          <TemplateStudioSubHeaderLeftView />
        </TestWrapper>
      </TemplateContext.Provider>
    )

    const editBtn = container.querySelector('[data-icon="Edit"]') as Element
    expect(editBtn).toBeDefined()
    await act(async () => {
      fireEvent.click(editBtn)
    })
    let modal = findDialogContainer()
    expect(modal).toMatchSnapshot()

    const cancelBtn = getByRole(modal!, 'button', { name: 'cancel' })
    await act(async () => {
      fireEvent.click(cancelBtn)
    })
    expect(templateContextMock.updateTemplate).toBeCalledTimes(0)

    await act(async () => {
      fireEvent.click(editBtn)
    })
    modal = findDialogContainer()
    const saveBtn = getByRole(modal!, 'button', { name: 'save' })
    await act(async () => {
      fireEvent.click(saveBtn)
    })
    expect(templateContextMock.updateTemplate).toBeCalled()
  })
})
