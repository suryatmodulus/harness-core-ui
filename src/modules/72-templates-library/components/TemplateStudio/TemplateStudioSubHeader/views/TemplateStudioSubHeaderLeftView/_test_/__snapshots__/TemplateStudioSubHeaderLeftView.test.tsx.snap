// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateStudioSubHeaderLeftView /> tests edit btn should work as expected 1`] = `
<div
  class="bp3-dialog createTemplateDialog"
>
  <div
    class="Layout--horizontal StyledProps--main"
  >
    <div
      class="StyledProps--font StyledProps--main basicDetails StyledProps--background-form StyledProps--padding-huge"
      style="width: 55%;"
    >
      <p
        class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey800 StyledProps--font-weight-bold StyledProps--font-size-medium StyledProps--margin-bottom-xlarge"
      >
        templatesLibrary.createNewModal.editHeading
      </p>
      <div
        class="OverlaySpinner--overlaySpinner"
      >
        <form
          class="FormikForm--main"
        >
          <div>
            <div
              class="Layout--vertical Layout--layout-spacing-huge StyledProps--main"
            >
              <div
                class="StyledProps--font StyledProps--main"
              >
                <div
                  class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                >
                  <div
                    class="StyledProps--font StyledProps--main"
                  >
                    <div
                      class="Layout--vertical StyledProps--main"
                    >
                      <div
                        class="StyledProps--font StyledProps--main main nameIdDescriptionTags"
                      >
                        <div
                          class="InputWithIdentifier--txtNameContainer TextInput--main"
                        >
                          <div
                            class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
                          >
                            <div
                              class="Text--withDocsTooltip"
                              data-tooltip-id="createStepTemplate_identifier"
                            >
                              <p
                                class="StyledProps--font StyledProps--main"
                              >
                                ID
                              </p>
                            </div>
                            <p
                              class="StyledProps--font StyledProps--main"
                            >
                              :
                            </p>
                            <span
                              class="bp3-popover-wrapper"
                            >
                              <span
                                class="bp3-popover-target"
                              >
                                <div
                                  class="bp3-editable-text bp3-disabled InputWithIdentifier--idValue"
                                >
                                  <span
                                    class="bp3-editable-text-content"
                                    style="height: 1px; line-height: 1px; min-width: 0;"
                                  >
                                    Test_Http_Template
                                  </span>
                                </div>
                              </span>
                            </span>
                          </div>
                          <div
                            class="bp3-form-group"
                            data-id="name-0"
                          >
                            <label
                              class="bp3-label"
                              for="name"
                            >
                              <span
                                class="Tooltip--acenter"
                                data-tooltip-id="createStepTemplate_name"
                              >
                                name
                              </span>
                               
                              <span
                                class="bp3-text-muted"
                              />
                            </label>
                            <div
                              class="bp3-form-content"
                            >
                              <div
                                class="bp3-input-group"
                              >
                                <input
                                  autocomplete="off"
                                  class="bp3-input"
                                  name="name"
                                  type="text"
                                  value="Test Http Template"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          class="StyledProps--font StyledProps--main"
                        >
                          <label
                            class="Label--main bp3-label descriptionLabel"
                            data-tooltip-id="createStepTemplate_description"
                          >
                            optionalField
                            <span
                              class="bp3-icon StyledProps--main editOpen"
                              data-icon="Edit"
                              data-name="edit"
                              data-testid="description-edit"
                            >
                              <svg
                                height="12"
                                viewBox="0 0 16 16"
                                width="12"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                          </label>
                        </div>
                        <div
                          class="StyledProps--font StyledProps--main"
                        >
                          <label
                            class="Label--main bp3-label descriptionLabel"
                            data-tooltip-id="createStepTemplate_tags"
                          >
                            optionalField
                            <span
                              class="bp3-icon StyledProps--main editOpen"
                              data-icon="Edit"
                              data-name="edit"
                              data-testid="tags-edit"
                            >
                              <svg
                                height="12"
                                viewBox="0 0 16 16"
                                width="12"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                          </label>
                        </div>
                      </div>
                      <div
                        class="bp3-form-group bp3-disabled"
                        data-id="versionLabel-1"
                      >
                        <label
                          class="bp3-label"
                          for="versionLabel"
                        >
                          <span
                            class="Tooltip--acenter"
                            data-tooltip-id="createStepTemplate_versionLabel"
                          >
                            templatesLibrary.createNewModal.versionLabel
                          </span>
                           
                          <span
                            class="bp3-text-muted"
                          />
                        </label>
                        <div
                          class="bp3-form-content"
                        >
                          <div
                            class="bp3-input-group bp3-disabled"
                          >
                            <input
                              autocomplete="off"
                              class="bp3-input"
                              disabled=""
                              name="versionLabel"
                              placeholder="templatesLibrary.createNewModal.versionPlaceholder"
                              type="text"
                              value="v1"
                            />
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="StyledProps--font StyledProps--main"
              >
                <div
                  class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-flex-end StyledProps--flex-justifyContent-flex-start"
                >
                  <button
                    class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-primary"
                    type="submit"
                  >
                    <span
                      class="bp3-button-text"
                    >
                      save
                    </span>
                  </button>
                  <button
                    class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
                    type="button"
                  >
                    <span
                      class="bp3-button-text"
                    >
                      cancel
                    </span>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div
      class="Layout--horizontal StyledProps--main preview StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-center StyledProps--padding-top-huge StyledProps--padding-bottom-huge"
    >
      <div
        class="StyledProps--font StyledProps--main container"
      >
        <div
          class="bp3-card bp3-elevation-0 Card--card templateCard"
        >
          <div />
          <div
            class="StyledProps--font StyledProps--main StyledProps--margin-right-small"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--margin-bottom-small StyledProps--flex"
            >
              <span
                class="bp3-icon StyledProps--main"
                data-icon="http-step"
              >
                <svg
                  height="24"
                  viewBox="0 0 32 32"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 27H2v-3H0v8h2v-3h3v3h2v-8H5v3zm4-1h2v6h2v-6h2v-2H9v2zm8 0h2v6h2v-6h2v-2h-6v2zm12-2h-4v8h2v-2h2a3 3 0 000-6zm0 4h-2v-2h2a1 1 0 010 2zM0 0v22h2V10h28v12h2V0H0zm30 8H2V2h28v6z"
                    fill="currentColor"
                  />
                  <path
                    d="M6 4H4v2h2V4zm4 0H8v2h2V4zm18 0H12v2h16V4z"
                    fill="currentColor"
                  />
                </svg>
              </span>
            </div>
          </div>
          <div
            class="StyledProps--font StyledProps--main"
          >
            <p
              class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-grey800"
              data-testid="Test_Http_Template"
              style="--text-line-clamp: 1;"
            >
              Test Http Template
            </p>
            <p
              class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-small StyledProps--color StyledProps--color-grey400 StyledProps--margin-top-xsmall"
              style="--text-line-clamp: 1;"
            >
              idLabel
            </p>
          </div>
          <div
            class="StyledProps--font StyledProps--main StyledProps--background-grey100"
            style="height: 1px;"
          />
          <div
            class="StyledProps--font StyledProps--main"
          >
            <span
              class="Tag--main"
            >
              <span
                class="bp3-tag version"
              >
                <span
                  class="bp3-text-overflow-ellipsis bp3-fill"
                >
                  v1
                </span>
              </span>
            </span>
          </div>
          <div
            class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-justifyContent-center StyledProps--padding-top-medium"
          >
            <div
              class="templateColor"
            >
              <div
                class="absolutePos"
              >
                <svg
                  fill="none"
                  height="18"
                  viewBox="0 0 121 18"
                  width="121"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M10.6577 4.36758L0.989019 17.5H120.044L111.076 4.58226C109.301 2.02513 106.385 0.5 103.272 0.5H18.3079C15.288 0.5 12.4481 1.93575 10.6577 4.36758Z"
                    fill="#EADEFF"
                    stroke="#E1D0FF"
                  />
                </svg>
              </div>
              <p
                class="StyledProps--font StyledProps--main text absolutePos"
                style="color: rgb(89, 43, 170);"
              >
                STEP
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main closeIcon Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
      type="button"
    >
      <span
        class="bp3-icon bp3-icon-cross StyledProps--main StyledProps--color StyledProps--color-grey500"
        icon="cross"
      >
        <svg
          data-icon="cross"
          height="24"
          viewBox="0 0 20 20"
          width="24"
        >
          <desc>
            cross
          </desc>
          <path
            d="M11.41 10l4.29-4.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71L10 8.59l-4.29-4.3a1.003 1.003 0 00-1.42 1.42L8.59 10 4.3 14.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4.29-4.3 4.29 4.29c.18.19.43.3.71.3a1.003 1.003 0 00.71-1.71L11.41 10z"
            fill-rule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;

exports[`<TemplateStudioSubHeaderLeftView /> tests snapshot test 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main subHeaderLeftView"
  >
    <div
      class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-right-medium StyledProps--flex StyledProps--flex-alignItems-center"
    >
      <div
        class="StyledProps--font StyledProps--main"
      >
        <div
          class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center"
        >
          <span
            class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-purple700"
            data-icon="template-library"
          >
            <svg
              height="20"
              viewBox="0 0 12 13"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
            >
              <defs>
                <path
                  d="M1.8 13c.2 0 .4-.3.4-.6V.6C2.2.3 2 0 1.8 0H.4C.2 0 0 .3 0 .6v11.7c0 .4.2.7.4.7h1.4zm8.8 0l.9-.3c.3-.1.5-.4.4-.8L8.9.6C8.8.3 8.5.1 8.1.2l-.9.3c-.4 0-.5.4-.5.7l3.1 11.3c.1.4.5.6.8.5zm-5 0c.2 0 .4-.3.4-.7V.6c0-.3-.2-.6-.4-.6H4.2c-.2 0-.4.3-.4.6v11.7c0 .4.2.7.4.7h1.4z"
                  id="template-library_svg__a"
                />
              </defs>
              <clippath
                id="template-library_svg__b"
              >
                <use
                  overflow="visible"
                  xlink:href="#template-library_svg__a"
                />
              </clippath>
              <path
                clip-path="url(#template-library_svg__b)"
                d="M-5-5h22v23H-5z"
                fill="#25a6f7"
              />
            </svg>
          </span>
          <span
            class="bp3-popover-wrapper"
          >
            <span
              class="bp3-popover-target Text--targetClass"
            >
              <p
                class="StyledProps--font StyledProps--main templateName StyledProps--color StyledProps--color-grey700 StyledProps--font-weight-bold"
                tabindex="0"
              >
                Test Http Template
              </p>
            </span>
          </span>
          <button
            class="bp3-button Button--button StyledProps--font StyledProps--main Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
            type="button"
          >
            <span
              class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey800"
              data-icon="Edit"
            >
              <svg
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
                <path
                  clip-rule="evenodd"
                  d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                  fill="currentColor"
                  fill-rule="evenodd"
                  opacity="0.3"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <div
        class="bp3-popover-wrapper DropDown--main versionDropDown"
      >
        <div
          class="bp3-popover-target"
        >
          <div
            class=""
          >
            <div
              class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder DropDown--minWidth StyledProps--flex"
              data-testid="dropdown-button"
              style="min-width: 130px;"
            >
              <div
                class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
              >
                <p
                  class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                  data-testid="dropdown-value"
                  style="--text-line-clamp: 1;"
                >
                  v1
                </p>
              </div>
              <div
                class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
              >
                <span
                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                  data-icon="main-chevron-down"
                >
                  <svg
                    height="8"
                    viewBox="0 0 14 8"
                    width="8"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                      fill="#4d5969"
                    />
                  </svg>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-link Button--size Button--small"
        type="button"
      >
        <span
          class="bp3-button-text"
        >
          common.setAsStable
        </span>
      </button>
    </div>
  </div>
</div>
`;

exports[`<TemplateStudioSubHeaderLeftView /> tests update stable template test 1`] = `
<div
  class="bp3-dialog useConfirmationDialog--dialog"
  style="width: 500px; min-height: 218px;"
>
  <div
    class="StyledProps--font StyledProps--main useConfirmationDialog--iconContainer StyledProps--flex"
  >
    <span
      class="bp3-icon StyledProps--main useConfirmationDialog--icon"
      data-icon="main-close"
    >
      <svg
        height="8"
        viewBox="0 0 14 14"
        width="8"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.6 7l5-5c.3-.2.4-.5.4-.8 0-.7-.5-1.2-1.2-1.2-.3 0-.6.1-.8.3l-5 5-5-5C1.8.1 1.5 0 1.2 0 .5 0 0 .5 0 1.2c0 .3.1.6.3.8l5 5-5 5c-.2.2-.3.5-.3.8 0 .7.5 1.2 1.2 1.2.3 0 .6-.1.8-.3l5-5 5 5c.2.2.5.4.8.4.6 0 1.2-.5 1.2-1.2 0-.3-.1-.6-.3-.8z"
          fill="#4d5969"
        />
      </svg>
    </span>
  </div>
  <div
    class="Layout--horizontal StyledProps--main useConfirmationDialog--header StyledProps--padding-left-xsmall"
  >
    <span
      class="bp3-icon StyledProps--main StyledProps--margin-right-small"
      data-icon="info-messaging"
    >
      <svg
        fill="none"
        height="32"
        viewBox="0 0 32 32"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M16 29.004c-7.18 0-13-5.82-13-13s5.82-13 13-13 13 5.82 13 13-5.82 13-13 13zm-1.499-18.5a1.5 1.5 0 103 0 1.5 1.5 0 00-3 0zm1.5 3.474a1 1 0 00-1 1v6.002a1 1 0 102 0v-6.002a1 1 0 00-1-1z"
          fill="#5452F6"
          fill-rule="evenodd"
        />
        <path
          clip-rule="evenodd"
          d="M16 28.004c-6.627 0-12-5.373-12-12s5.373-12 12-12 12 5.373 12 12-5.373 12-12 12zm-.244-15.011a2.5 2.5 0 11.49 0 2 2 0 011.755 1.985v6.002a2 2 0 01-4 0v-6.002a2 2 0 011.755-1.985zM3 16.004c0 7.18 5.82 13 13 13s13-5.82 13-13-5.82-13-13-13-13 5.82-13 13zm13.001-4a1.5 1.5 0 110-3 1.5 1.5 0 010 3zm-1 2.974a1 1 0 112 0v6.002a1 1 0 11-2 0v-6.002z"
          fill="#5452F6"
          fill-rule="evenodd"
        />
        <path
          d="M27.598 28.94c.83.358 1.672-.483 1.314-1.314L25.99 20.85a1 1 0 00-1.625-.311l-3.854 3.854a1 1 0 00.31 1.625l6.777 2.922z"
          fill="#5452F6"
        />
      </svg>
    </span>
    <p
      class="StyledProps--font StyledProps--main StyledProps--font-variation-h4"
    >
      templatesLibrary.setAsStableTitle
    </p>
  </div>
  <div
    class="Layout--vertical StyledProps--main useConfirmationDialog--body StyledProps--font StyledProps--font-variation-body StyledProps--color StyledProps--color-black StyledProps--margin-top-large StyledProps--margin-bottom-xxlarge"
  >
    templatesLibrary.setAsStableText
  </div>
  <div
    class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
  >
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        confirm
      </span>
    </button>
    <button
      class="bp3-button Button--button StyledProps--font StyledProps--main Button--with-current-color Button--variation Button--variation-tertiary"
      type="button"
    >
      <span
        class="bp3-button-text"
      >
        cancel
      </span>
    </button>
  </div>
</div>
`;
